GraphQL playground page (#2633)
umbraco/UmbracoDocs: Umbraco-Heartcore/Getting-Started-Cloud/GraphQL-Playground/index.md

---
versionFrom: 8.0.0
meta.Title: "Umbraco Heartcore GraphQL Playground"
meta.Description: "Documentation for Umbraco Heartcore GraphQL Playground"
---

# GraphQL Playground

With GraphQL query language, you will be able to limit the amount of data transferred to and from your Heartcore instance by specifying exactly what kind of information you would like to get in your API calls.

You can test and build GraphQL queries directly in the backoffice-integrated GraphQL Playground IDE.

In the Settings section in the Umbraco Backoffice, you will find the Headless tree. From there you can use the GraphQL playground to test your queries against your project's schema.

The Playground comes with basic features such as real-time error highlighting, syntax highlighting, formatting, query history and more.

## Accessing GraphQL Playground

1. Log into the project backoffice
2. Navigate to the Settings dashboard
3. Find the Headless tree near the bottom of the navigation list
4. Click on GraphQL Playground to open the IDE

![Where to find GraphQL Playground](images/where-to-find-graphql-playground.png)

> From there, you can start querying as soon as you have some content set up!

Reviewer: ```suggestion
From there, you can start querying as soon as you have some content set up.
```

> # Features
> 
> ## Documentation
> 

Reviewer: Try to not use H1 more than once in an article.

```suggestion
## Features

### Documentation

```

GraphQL Playground's built-in documentation can be found on the right side of the IDE, under the "Docs" tab.

The documentation tab grants a quick overview of how content is structured using the document types in your solution. It also gives an idea on how to access specific properties each document type has, as well as their properties.

![Documentation Tab in the Playground](images/docs-helper.png)

The documentation will be automatically updated as you add new document types to match the schema you have on your project.

> 

Reviewer: Would it make sense here, to link to the API Docs that are being written on GraphQL?

Author: Yes.
Will do that.

## Schema

> The Schema tab also can be found on the right side of the IDE. It is located below the "Docs" tab.

Reviewer: ```suggestion
The Schema tab can also be found on the right side of the IDE. It is located below the "Docs" tab.
```

Author: That was my first though as well, however I decided to move the **also** before **can** - since in the context of this section, I want to say "The schema tab, like the Docs tab, can ALSO be found on the right side". Saying "The Schema tab can also be found on the right side" sounds like that tab can be found in one other location.

The Schema menu explains the relationships between specific objets and properties in your solution, and will be updated automatically as you make changes to the project's schema - document types, datatypes, and so on.

![Schema Tab in the Playground](images/schema-helper.png)

## Running GraphQL queries

> To run a query, on the left side of the IDE type your query - could be something as basic as getting the names of all your content pages:

Reviewer: ```suggestion
To run a query use the left side of the IDE to type in your query - could be something as basic as getting the names of all your content pages:
```

```json
{
  allContent {
    items {
      name
    }
  }
}
```

:::note
At the moment, the Backoffice GraphQL Playground allows only for queries - other operation types, that is mutations and subscriptions, are not supported.
:::

After that, press the "Play" button sitting in the center of the Playground. If the query is constructed correctly, the results will be loaded into the right section of the IDE.

![Schema Tab in the Playground](images/sample-query.png)

> If you are satisfied with the result, you can copy the client URL (CURL) by clicking the `COPY CURL` button located in the top part of the Playground, next to the address bar.

Reviewer: ```suggestion
When you are satisfied with the result, you can copy the client URL (CURL) by clicking the `COPY CURL` button located in the top part of the Playground, next to the address bar.
```

---
versionFrom: 8.0.0
meta.Title: "Umbraco Heartcore GraphQL Playground"
meta.Description: "Documentation for Umbraco Heartcore GraphQL Playground"
---

# GraphQL Playground

> With GraphQL query language, you will be able to limit the amount of data transferred to and from your Heartcore instance by specifying exactly what kind of information you would like to get in your API calls.

Reviewer 2: ```suggestion
With GraphQL query language, you will be able to limit the amount of data transferred to and between your Heartcore instance and the client by specifying exactly what kind of information you would like to get in your API calls.
```

> With GraphQL query language, you will be able to limit the amount of data transferred to and between your Heartcore instance and the client by specifying exactly what kind of information you would like to get in your API calls.

Reviewer 3: With the GraphQL query language

Reviewer 3: data transferred to and from your Heartcore

Author: Thanks, added your suggestions in a new commit

> You can test and build GraphQL queries directly in the backoffice-integrated GraphQL Playground IDE.

Reviewer 2: ```suggestion
You can test and build GraphQL queries directly in the backoffice-integrated GraphQL Playground.
```

> In the Settings section in the Umbraco Backoffice, you will find the Headless tree. From there you can use the GraphQL playground to test your queries against your project's schema.

Reviewer 2: ```suggestion
In the Settings section in the Umbraco Backoffice, you will find the Headless tree. From there you can use the GraphQL Playground to test your queries against your project's schema.
```

The Playground comes with basic features such as real-time error highlighting, syntax highlighting, formatting, query history and more.

## Accessing GraphQL Playground

1. Log into the project backoffice
2. Navigate to the Settings dashboard
3. Find the Headless tree near the bottom of the navigation list

> 4. Click on GraphQL Playground to open the IDE

Reviewer 2: ```suggestion
4. Click on GraphQL Playground
```

![Where to find GraphQL Playground](images/where-to-find-graphql-playground.png)

> From there, you can start querying as soon as you have some content set up.

Reviewer 3: From there, you can start querying as soon as your content is published.

## Features

### Documentation

> GraphQL Playground's built-in documentation can be found on the right side, under the "Docs" tab.

Reviewer 3: The GraphQL Playground's

> GraphQL Playground's built-in documentation can be found on the right side of the IDE, under the "Docs" tab.

Reviewer 2: ```suggestion
GraphQL Playground's built-in documentation can be found on the right side, under the "Docs" tab.
```

> The documentation tab grants a quick overview of how content is structured using the Document Types in your solution. It also gives an idea on how to access specific properties each Document Type has, as well as their properties.

Reviewer 3: It also gives an idea of how to access specific properties on each of the Document Types as well as nested properties and properties from Compositions.

> The documentation tab grants a quick overview of how content is structured using the document types in your solution. It also gives an idea on how to access specific properties each document type has, as well as their properties.

Reviewer 2: ```suggestion
The documentation tab grants a quick overview of how content is structured using the Document Types in your solution. It also gives an idea on how to access specific properties each Document Type has, as well as their properties.
```

![Documentation Tab in the Playground](images/docs-helper.png)

> The documentation will be automatically updated as you add new document types to match the schema you have on your project.

Reviewer 2: ```suggestion
The documentation will be automatically updated as you add new Document Types to match the schema you have on your project.
```

## Schema

> The Schema tab also can be found on the right side of the IDE. It is located below the "Docs" tab.

Reviewer 2: ```suggestion
The Schema tab also can be found on the right side of the GraphQL Playground. It is located below the "Docs" tab.
```

> The Schema tab also can be found on the right side of the GraphQL Playground. It is located below the "Docs" tab.

Reviewer 3: The Schema tab can also be found

> The Schema menu explains the relationships between specific objets and properties in your solution, and will be updated automatically as you make changes to the project's schema - document types, datatypes, and so on.

Reviewer 2: ```suggestion
The Schema menu explains the relationships between specific objets and properties in your solution, and will be updated automatically as you make changes to the project's schema - Document Types, Data Types, and so on.
```

> The Schema menu explains the relationships between specific objets and properties in your solution, and will be updated automatically as you make changes to the project's schema - Document Types, Data Types, and so on.

Reviewer 3: make changes to the project's schema meaning Document Types, Data Types, Document Type Compositions and Document Type Elements.

![Schema Tab in the Playground](images/schema-helper.png)

## Running GraphQL queries

> To run a query use the left side of the IDE to type in your query - could be something as basic as getting the names of all your content pages:

Reviewer 2: ```suggestion
To run a query use the left side of the GraphQL Playground to type in your query - could be something as basic as getting the names of all your content pages:
```

```json
{
  allContent {
    items {
      name
    }
  }
}
```

To learn more about querying, visit our [API documentation for GraphQL](../../API-Documentation/GraphQL/index.md).

:::note
At the moment, the Backoffice GraphQL Playground allows only for queries - other operation types, that is mutations and subscriptions, are not supported.
:::

> After that, press the "Play" button sitting in the center of the Playground. If the query is constructed correctly, the results will be loaded into the right section of the IDE.

Reviewer 2: ```suggestion
After that, press the "Play" button sitting in the center of the Playground. If the query is constructed correctly, the results will be loaded into the right section.
```
